src/Control/Lens/Cons.hs

```diff
@@ -319,6 +323,42 @@ _tail :: Cons s s a a => Traversal' s s
 _tail = _Cons._2
 {-# INLINE _tail #-}
 
+-- | Convert one type with a 'Cons' instance into the other.
+--
+-- Rewrite rules are provided for efficient conversion between
+-- 'String' and 'Text', @['Word8']@ and 'ByteString', and lazy and
+-- strict 'Text' and 'ByteString'.  Programs must be compiled
+-- with @-O@ to use them.
```

Comment:
I agree. But I just don't like this solution. There is e.g. `https://hackage.haskell.org/package/string-conversions-0.4.0.1/docs/Data-String-Conversions.html#v:convertString` which though assumes the encoding of `ByteString`, so isn't problematic either.